docs/halibut-protocol.md

```diff
@@ -0,0 +1,66 @@
+# Halibut Protocol data exchange
+
+This document outlines the data that is exchanged between the **Client** and **Service**. In this document:
+
+ - **Client** is what makes the remote procedure calls e.g. Octopus Server.
+ - **Service** is what executes the methods e.g. Tentacle. 
+
+
+When invoking a remote method in both polling and listening mode, the same general process is followed:
+
+1.  After the TCP connection is intiated, both server and client identify themselves to each other through a set of control messages.
```

Comment:
It might be worth mentioning that the identification is different for listening vs polling tentacles and adding a link to those sections.

Reply:
Done